src/query/catalog/src/plan/pushdown.rs

```diff
@@ -29,6 +31,18 @@ pub struct PrewhereInfo {
     pub filter: Expression,
 }
 
+#[derive(serde::Serialize, serde::Deserialize, Clone, Debug, PartialEq, Eq)]
+pub struct StageInfo {
```

Comment:
About `StageContext`? And I prefer to rename `UserStageInfo` into `StageInfo` instead.

Reply:
This is used for pushdown, I changed the name to `StagePushDownInfo` but not `StageContext`.
BTW, `UserStageInfo` is better than `StageInfo` here, because in distributed execute system, there is a `Stage`, `StageInfo` may be confused for them.